autogpt_platform/backend/backend/blocks/google/_drive.py

```diff
@@ -49,144 +41,115 @@ class GoogleDriveFile(BaseModel):
-        ...             allowed_views=["DOCUMENTS"],
+        ...     class Input(BlockSchemaInput):
+        ...         spreadsheet: GoogleDriveFile = GoogleDriveFileField(
+        ...             title="Select Spreadsheet",
+        ...             credentials_kwarg="creds",
+        ...             allowed_views=["SPREADSHEETS"],
         ...         )
         ...
-        ...         files: list[GoogleDriveFile] = GoogleDrivePickerField(
-        ...             title="Select Multiple Files",
-        ...             multiselect=True,
-        ...             allow_folder_selection=True,
-        ...         )
+        ...     async def run(
+        ...         self, input_data: Input, *, creds: GoogleCredentials, **kwargs
+        ...     ):
+        ...         # creds is automatically populated
+        ...         file = input_data.spreadsheet
     """
-    # Build configuration that will be sent to frontend
+
+    # Determine scopes - drive.file is sufficient for picker-selected files
+    scopes = credentials_scopes or ["https://www.googleapis.com/auth/drive.file"]
+
+    # Build picker configuration with auto_credentials embedded
     picker_config = {
-        "multiselect": multiselect,
-        "allow_folder_selection": allow_folder_selection,
+        "multiselect": False,
+        "allow_folder_selection": False,
         "allowed_views": list(allowed_views) if allowed_views else ["DOCS"],
+        "scopes": scopes,
+        # Auto-credentials config tells frontend to include _credentials_id in output
+        "auto_credentials": {
+            "provider": "google",
+            "type": "oauth2",
+            "scopes": scopes,
+            "kwarg_name": credentials_kwarg,
+        },
     }
 
-    # Add optional configurations
     if allowed_mime_types:
         picker_config["allowed_mime_types"] = list(allowed_mime_types)
 
-    # Determine required scopes based on config
-    base_scopes = scopes if scopes is not None else []
-    picker_scopes: set[str] = set(base_scopes)
-    if allow_folder_selection:
-        picker_scopes.add("https://www.googleapis.com/auth/drive")
-    else:
-        # Use drive.file for minimal scope - only access files selected by user in picker
-        picker_scopes.add("https://www.googleapis.com/auth/drive.file")
-
-    picker_config["scopes"] = sorted(picker_scopes)
-
-    # Set appropriate default value
-    default_value = [] if multiselect else None
-
-    # Use SchemaField to handle format properly
     return SchemaField(
-        default=default_value,
+        default=None,
         title=title,
         description=description,
-        placeholder=placeholder or "Choose from Google Drive",
+        placeholder=placeholder or "Select from Google Drive",
+        # Use google-drive-picker format so frontend renders existing component
         format="google-drive-picker",
         advanced=False,
         json_schema_extra={
             "google_drive_picker_config": picker_config,
```

Comment:
The need of creating a schema for this is increasing, I've started losing track what fields are required for this and it's used in several places.